pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/ClientCachedDao.java

```diff
@@ -0,0 +1,119 @@
+        String auth0ClientId;
+        String auth0Domain;
+
+        StudyClientConfigCacheKey(String auth0ClientId, String auth0Domain) {
+            this.auth0ClientId = auth0ClientId;
+            this.auth0Domain = auth0Domain;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+
+            StudyClientConfigCacheKey that = (StudyClientConfigCacheKey) o;
+
+            if (auth0ClientId != null ? !auth0ClientId.equals(that.auth0ClientId) : that.auth0ClientId != null) {
+                return false;
+            }
+            return auth0Domain != null ? auth0Domain.equals(that.auth0Domain) : that.auth0Domain == null;
+        }
+
+        @Override
+        public int hashCode() {
+            int result = auth0ClientId != null ? auth0ClientId.hashCode() : 0;
+            result = 31 * result + (auth0Domain != null ? auth0Domain.hashCode() : 0);
+            return result;
+        }
+    }
+
+    private static Cache<StudyClientConfigCacheKey, StudyClientConfiguration> CLIENT_STUDY_CONFIG_CACHE;
+
+    private static void initCache() {
+        if (CLIENT_STUDY_CONFIG_CACHE == null || CLIENT_STUDY_CONFIG_CACHE.isClosed()) {
+            CLIENT_STUDY_CONFIG_CACHE = CacheService.getInstance().getOrCreateCache(CACHE_NAME, new Duration(MINUTES, 10),
+                    ClientCachedDao.class);
+        }
+    }
+
+    public ClientCachedDao(Handle handle) {
+        super(handle, ClientDao.class);
+        initCache();
+    }
+
+    @Override
+    public JdbiClient getClientDao() {
+        return target.getClientDao();
+    }
+
+    @Override
+    public JdbiUmbrellaStudy getUmbrellaStudyDao() {
+        return target.getUmbrellaStudyDao();
+    }
+
+    @Override
+    public JdbiClientUmbrellaStudy getClientUmbrellaStudyDao() {
+        return target.getClientUmbrellaStudyDao();
+    }
+
+    @Override
+    public List<ClientDto> findAllPermittedClientsForStudy(long studyId) {
+        return target.findAllPermittedClientsForStudy(studyId);
+    }
+
+    @Override
+    public long registerClient(String auth0ClientId, String auth0ClientSecret, Collection<String> studyGuidsToAccess,
+                               String encryptionKey, Long auth0TenantId) {
+        return target.registerClient(auth0ClientId, auth0ClientSecret, studyGuidsToAccess, encryptionKey, auth0TenantId);
+    }
+
+    @Override
+    public StudyClientConfiguration getConfiguration(String auth0ClientId, String auth0Domain) {
+        var key = new StudyClientConfigCacheKey(auth0ClientId, auth0Domain);
+        var config = CLIENT_STUDY_CONFIG_CACHE.get(key);
+        if (config == null) {
+            config = target.getConfiguration(auth0ClientId, auth0Domain);
+            CLIENT_STUDY_CONFIG_CACHE.put(key, config);
+        }
+        return config;
+    }
```

Comment:
We finally use the cache!
Note that the first call will always result  in a query.